src/html.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { ReactNode } from 'react';
+import cheerio from 'cheerio';
+import { Stream } from 'stream';
+/**
+ *
+ * Gatsby does a two-pass render for HTML. It loops through your pages first
+ * rendering only the body and then takes the result body HTML string and
+ * passes it as the `body` prop to this component to complete the render.
+ */
+
+const postProcess = (body: string) => {
+  let globalStyles = '';
+
+  if (body.includes('<template')) {
+    const $ = cheerio.load(body);
+    const templates = $('template');
+
+    templates.toArray().forEach(element => {
+      const str = $(element).html();
+      const styles = cheerio.load(String(str))('style');
+      globalStyles += styles
+        .toArray()
+        .map(el => $(el).html())
+        .join(' ');
+    });
+  }
+  return { body, globalStyles };
+};
+
+interface HTMLProps {
+  htmlAttributes: {};
+  headComponents: ReactNode[];
+  bodyAttributes: {};
+  preBodyComponents: ReactNode[];
+  body: string;
+  postBodyComponents: ReactNode[];
```

Comment:
`ReactNode` can also be arrays; isn't enough just use `ReactNode` directly?
```suggestion
  headComponents: ReactNode;
  bodyAttributes: {};
  preBodyComponents: ReactNode;
  body: string;
  postBodyComponents: ReactNode;
```

Reply:
following the ejected types from gatsby https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/cache-dir/default-html.js#L31